images/operator/Dockerfile

```diff
@@ -69,3 +71,35 @@ COPY --from=builder /out/${TARGETOS}/${TARGETARCH}/LICENSE.all /LICENSE.all
 WORKDIR /
 ENV GOPS_CONFIG_DIR=/
 CMD ["/usr/bin/cilium-${OPERATOR_VARIANT}"]
+
+# Intermediate stage because expansion is not working in COPY --from
+FROM --platform=${BUILDPLATFORM} ${CILIUM_BUILDER_IMAGE} as debug-builder
+
+#
+# Cilium debug image.
+#
+# Typical image bulids will stop above at the 'release' target, but
+# developers follow this Dockerfile to the end. Starting from a release
+# image, install delve debugger and wrap the cilium-operator binary calls
+# with a script that automatically provisions the debugger on a
+# dedicated port.
+FROM --platform=${BUILDPLATFORM} ${CILIUM_RUNTIME_IMAGE} as debug
```

Comment:
`release` is `scratch` in case of the operator - therefore not containing `bash` - which is necessary for the `debug-wrapper`. 

that's the reason why i decided to rebuild the debug image on top of the runtime image - which brings the duplication.

another option would be change `release` (don't prefer this one) or copying `bash` into the scratch image (if this is possible).

Reply:
I'd be open to replacing debug-wrapper.sh if you have a good idea. My main motivations with the current approach were just to change as few things as possible from the existing release images&helm charts.

It didn't occur to me to statically link the binary against delve to introduce the debugger, the wrapper script was just the minimal way to get that done.

Reply:
* replaced the bash script with a small and simple go app which gets built in the builder image
* had to install delve with `CGO=0`

with this changes in place, it possible to build the debug image of the operator on top of its release image and only add the necessary stuff (copy delve, copy debug wrapper, re-adding operator binary itself (because `mv` isn't available in `scratch` too))

this way the diff between `release` & `debug` and `agent` & `operator` are minimal even though the operator is built on top of `scratch`